### ISSUE TITLE
[Platform Compat Analyzer] Attributes are not respected on event add/remove methods

### ISSUE THREAD

**Issue by jeffhandley** [dotnet/roslyn-analyzers#4072]
### Describe the bug

Diagnostics are not raised when code registers or unregisters event handlers and the `add` or `remove` methods have `[SupportedOSPlatform]` or `[UnsupportedOSPlatform]` attributes applied.  When the attribute is applied to the `event` itself, diagnostics are raised as expected.

### Steps To Reproduce

```csharp
        [SupportedOSPlatform("windows")] // References to this event DO report diagnostics as expected
        public static event EventHandler WindowsOnlyEvent
        {
            add { }
            remove { }
        }

        public static event EventHandler WindowsOnlyEventAdd
        {
            [SupportedOSPlatform("windows")] // References to register an event handler DO NOT report diagnostics as expected
            add { }
            remove { }
        }

        public static event EventHandler WindowsOnlyEventRemove
        {
            add { }

            [SupportedOSPlatform("windows")] // References to unregister an event handler DO NOT report diagnostics as expected
            remove { }
        }
```
### Expected behavior

Annotations on event `add` / `remove` methods should be respected and should result in diagnostics when referenced in code that is reachable in unsupported platforms.

### Actual behavior

No diagnostic is raised.

### Analyzer

**Platform Compatibility Analyzer** (local RC1 build)

**Diagnostic ID**: [CA1416](https://docs.microsoft.com/visualstudio/code-quality/ca1416)

## Additional context

We do not have any occurrences of this pattern for the APIs we've marked with attributes, and I'm not certain this was expected to work.  We should consult with @terrajobst to determine if this should be fixed.

**Comment by terrajobst** [comment]
This isn't as bad as #4071 (I'm not aware of a single event we attributed) but it's good hygiene. We'd expect this to "just work".

**Comment by buyaa-n** [comment]
> On second thought, while we haven't attributed events, we have attributed types, some of which have events. I'm now wondering whether we correctly warn for SomeType.SomEvent += somethod.

Yes we are, it was not working only for attributes applied to the `add/remove` accessors
